core/Enclave/enclave_hsm.cpp

```diff
@@ -458,27 +540,111 @@ sgx_status_t enclave_rsa_encrypt(const uint8_t *cmk_blob,
                                  uint8_t keyspec,
                                  uint8_t padding_mode)
 {
-    const rsa_key_data_t *temp_pub_key = (rsa_key_data_t *)cmk_blob;
-
-    BIO *bio = BIO_new_mem_buf(temp_pub_key->rsa_public_key, -1);
+    sgx_status_t ret = SGX_ERROR_UNEXPECTED;
 
-    RSA* rsa = PEM_read_bio_RSAPublicKey(bio, &rsa, NULL, g_domain_key);
+    if (keyspec == NULL || padding_mode == NULL)
+        return SGX_ERROR_INVALID_PARAMETER;
 
-    if(rsa == NULL)
+    uint32_t sealed_rsa_len;
+    switch(keyspec)
     {
-        printf("failed to load rsa key\n");
+        case EH_RSA_2048:
+            sealed_rsa_len = sgx_calc_gcm_data_size(0, 2105);
+
+            if (plaintext_len > RSA_OAEP_2048_SHA_256_MAX_ENCRYPTION_SIZE) {
+                printf("ecall rsa_encrypt plain len is up to 190B.\n");
```

Comment:
seems that some place you use the function to get the max size, some place use the MACROs